tests/support/Pipelines/EccentricityControl/Test_EccentricityControl.py

```diff
@@ -0,0 +1,171 @@
+
+        # Linear in eccentricity orbit approximation
+        def r(t):
+            return r_bar * (1 - ecc * np.cos(Omega_bar * t))
+
+        def phase(t):
+            return (
+                Omega_bar * t
+                + (2.0 * ecc / Omega_bar) * np.sin(Omega_bar * t)
+                + initial_phase
+            )
+
+        # Define the spirals for equal mass orbits
+        def SpiralA(t):
+            return np.array(
+                [
+                    x0 + 0.5 * r(t) * np.cos(phase(t)),
+                    y0 + 0.5 * r(t) * np.sin(phase(t)),
+                    z0 + z1 * (1 - 0.1) * t,
+                ]
+            )
+
+        def SpiralB(t):
+            return np.array(
+                [
+                    x0 - 0.5 * r(t) * np.cos(phase(t)),
+                    y0 - 0.5 * r(t) * np.sin(phase(t)),
+                    z0 + z1 * (1 - 0.1) * t,
+                ]
+            )
+
+        # Generate time samples
+        tTable = np.arange(0, (nsamples + 1) * dt, dt)
+
+        # Map time to spiral points
+        AhA_data = np.array([[t, *SpiralA(t), *SpiralA(t)] for t in tTable])
+        AhB_data = np.array([[t, *SpiralB(t), *SpiralB(t)] for t in tTable])
+
+        with spectre_h5.H5File(self.h5_filename, "w") as h5_file:
+            # Insert dataset for AhA
+            dataset_AhA = h5_file.insert_dat(
+                "ApparentHorizons/ControlSystemAhA_Centers.dat",
+                legend=[
+                    "Time",
+                    "GridCenter_x",
+                    "GridCenter_y",
+                    "GridCenter_z",
+                    "InertialCenter_x",
+                    "InertialCenter_y",
+                    "InertialCenter_z",
+                ],  # Legend for the dataset
+                version=0,  # Version number
+            )
+            # Populate dataset with AhA data
+            for data_point in AhA_data:
+                dataset_AhA.append(data_point)
+            # Close dataset for AhA
+            h5_file.close_current_object()
+
+            # Insert dataset for AhB
+            dataset_AhB = h5_file.insert_dat(
+                "ApparentHorizons/ControlSystemAhB_Centers.dat",
+                legend=[
+                    "Time",
+                    "GridCenter_x",
+                    "GridCenter_y",
+                    "GridCenter_z",
+                    "InertialCenter_x",
+                    "InertialCenter_y",
+                    "InertialCenter_z",
+                ],  # Legend for the dataset
+                version=0,  # Version number
+            )
+            # Populate dataset with AhB data
+            for data_point in AhB_data:
+                dataset_AhB.append(data_point)
+            # Close dataset for AhB
+            h5_file.close_current_object()
+        logging.info(
+            f"Successfully created and populated HDF5 file: {self.h5_filename}"
+        )
+
+    def test_cli(self):
```

Comment:
You have this test with the mock data set up, but don't actually check if the returned eccentricity and param updates make sense. Could you add a test function that invokes the non-CLI function and check the return values are reasonable?